spec/controllers/api/v4/analytics/overdue_lists_controller_spec.rb

```diff
@@ -0,0 +1,40 @@
+require "rails_helper"
+
+RSpec.describe Api::V4::Analytics::OverdueListsController, type: :controller do
+  let(:request_user) { create(:user) }
+  let(:request_facility) { create(:facility, facility_group: request_user.facility.facility_group) }
+
+  describe "#show" do
+    before :each do
+      request.env["HTTP_X_USER_ID"] = request_user.id
```

Comment:
Given how many times we do this in our specs, would it be worth extracting this setup into a helper? We seem to have a lot of code like this:
```
        let(:request_user) { FactoryBot.create(:user) }
        let(:request_facility) { FactoryBot.create(:facility, facility_group: request_user.facility.facility_group) }
        let(:HTTP_X_USER_ID) { request_user.id }
        let(:HTTP_X_FACILITY_ID) { request_facility.id }
        let(:Authorization) { "Bearer #{request_user.access_token}" }
        let(:Accept) { "text/html" }
```

I'm not suggesting you do this as part of this PR. It's just a thought.

Reply:
That sounds like a good idea, can defo refactor in a future PR.